subsys/net/lib/lwm2m/lwm2m_message_handling.c

```diff
@@ -64,6 +64,9 @@ LOG_MODULE_REGISTER(LOG_MODULE_NAME);
 #ifdef CONFIG_LWM2M_RW_SENML_CBOR_SUPPORT
 #include "lwm2m_rw_senml_cbor.h"
 #endif
+#ifdef CONFIG_LWM2M_GATEWAY_OBJ_SUPPORT
+#include "lwm2m_obj_gateway.h"
+#endif
```

Comment:
```suggestion
#include "lwm2m_obj_gateway.h"
```

This can be unconditional. the `IS_ENABLED()`  doesn't call the API if GW object is not enabled, but it needs the header for compiler to check the call.